config and fields libs Update

KenshiDRK opened the pull request:
Updated config lib to solve an error if t_merge was nil and updated the
packet 0x027 on fields.lua

Commit:
config and fields libs Update

Updated config lib to solve an error if t_merge was nil and updated the
packet 0x027 on fields.lua

z16 commented on addons/libs/config.lua:
```diff
@@ -142,114 +142,116 @@ function merge(t, t_merge, path)
         keys[tostring(key):lower()] = key
     end
+    
+    if t_merge then
```
This diff is awful to read due to the indentation. Was all you changed the if check? If so, I'd prefer for you to just use this:
```lua
    if not t_merge then
        return t
    end
```

KenshiDRK replied:
yea the if check is all I changed

z16 commented on addons/libs/packets/fields.lua:
```diff
@@ -1554,15 +1554,19 @@ fields.incoming[0x026] = L{
 }
 
--- Encumbrance Release
+local function mid(val)
```
I assume this should stand for "message ID". I find the name a bit odd. Personally I'd just use a function like this:
```lua
local function sub(amount, val)
    return val - amount
end
```

Then you can use it like so:
```lua
    {ctype='unsigned short',    label='Message ID',         fn=sub+{0x8000}},   -- 0A
```

This is more general and we could use it in other places as well (there are several packets that contain values with constants offsets). You might also add an `add` function that does the same but adding the amount.

I would also place the function at the top, along with the others. Take a look at [line 65](https://github.com/Windower/Lua/blob/dev/addons/libs/packets/fields.lua#L65).

KenshiDRK replied:
I'll make the changes as soon as possible

Commit:
config and fields libs Update

Updated config lib to solve an error if t_merge was nil and updated the
packet 0x027 on fields.lua
